substrateinterface/constants.py

```diff
@@ -39,3 +40,10 @@
         "default": "0x00000000"
     },
 }
+
+if os.getenv('GITHUB_REF'):
+    if not os.getenv('GITHUB_REF').startswith('refs/tags/v'):
+        raise ValueError('Incorrect tag format {}'.format(os.getenv('GITHUB_REF')))
+    __version__ = os.getenv('GITHUB_REF').replace('refs/tags/v', '')
+else:
+    __version__ = '0.0.0-dev1'
```

Comment:
not sure if it is useful for you but we use https://pypi.org/project/setuptools-scm/ with a fallback version that we bump with a script